test/runners.jl

```diff
@@ -153,6 +155,40 @@ end
         end
     end
 
+    # Checks that the compiler/linker include a build-id
+    # This is only available on ELF-based platforms
+    @testset "Compilation - build-id note $(platform) - $(compiler)" for platform in elf_platforms, compiler in ("cc", "gcc", "clang", "c++", "g++", "clang++")
+        mktempdir() do dir
+            ur = preferred_runner()(dir; platform=platform)
+            iobuff = IOBuffer()
+            test_c = """
+                #include <stdlib.h>
+                int test(void) {
+                    return 0;
+                }
+                """
+            test_script = """
+                set -e
+                # Make sure setting `CCACHE` doesn't affect the compiler wrappers.
+                export CCACHE=pwned
+                export USE_CCACHE=false
+                echo '$(test_c)' > test.c
+                # Build object file
+                $(compiler) -Werror -c test.c -o test.o
+                # Build shared library
+                $(compiler) -Werror -shared test.c -o libtest.\${dlext}
+
+                # Print out the notes in the library
+                readelf -n libtest.\${dlext}
+                """
+            cmd = `/bin/bash -c "$(test_script)"`
+            @test run(ur, cmd, iobuff)
+            seekstart(iobuff)
+            # Make sure the compiled library has the note section for the build-id
+            @test occursin(r".note.gnu.build-id", readchomp(iobuff))
```

Comment:
Probably would work without a regex (otherwise we'd want to escape the dots)?
```suggestion
            @test occursin(".note.gnu.build-id", readchomp(iobuff))
```